iotilecore/iotile/core/utilities/async_tools/event_loop.py

```diff
@@ -407,6 +407,19 @@ def inside_loop(self):
             if isinstance(result, Exception):
                 self._logger.error("Error stopping task %s: %s", task.name, repr(result))
 
+        # In some cases, tasks may not properly register or cancel their subtasks.
+        # This catches that in development, and also in deployment when there are third-party libraries
+        # that have this fault (such as hbmqtt)
+        tasks = [task for task in asyncio.Task.all_tasks() if task is not
+                 asyncio.tasks.Task.current_task() and not task.done()]
+        list(map(lambda task: task.cancel(), tasks))
```

Comment:
I don't have a huge issue with this line but I've never seen it written that way before.  A list comprehension would also do it:

```python
[task.cancel() for task in tasks]
```